src/main/java/org/opensearch/geospatial/GeospatialParser.java

```diff
@@ -7,19 +7,24 @@
 
 import java.util.HashMap;
 import java.util.Map;
+import java.util.Objects;
+
+import org.opensearch.common.bytes.BytesReference;
+import org.opensearch.common.xcontent.XContentHelper;
+import org.opensearch.common.xcontent.XContentType;
 
 /**
  * GeospatialParser provides helper methods to parse/extract/transform input to
  * desired formats.
  */
-public class GeospatialParser {
+public final class GeospatialParser {
```

Comment:
why we need final here?

Reply:
To make sure no one can override this .

Reply:
This statement is to question :D 
I mean is there a reason why it has been added just now, not with initial change?

Reply:
i missed it in my previous commit :D . Do you prefer separate commit or PR?

Reply:
nope that's fine, just want to understand the reason, thank you!